### PR TITLE
GPC-272: IP restrict admin access

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

File: terraform/dev/main.tf [hunk 1]
```diff
@@ -106,6 +106,8 @@ module "data_share_service" {
 
   admin_alerts_enabled           = false
   database_tunnel_alerts_enabled = false
+
+  admin_login_allowed_ip_blocks = null
```
Comment (oskarwilliams) [inline terraform/dev/main.tf:110]: can we not just provide a default and not set it here?

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by oskarwilliams — APPROVED** [review]

File: terraform/dev/main.tf (hunk 1 above)
Comment (ethanmills) [reply terraform/dev/main.tf:110]: I'd like not restricting to be a conscious choice so we don't forget it in prod

File: terraform/demo/main.tf [hunk 2]
```diff
@@ -106,6 +106,25 @@ module "data_share_service" {
 
   admin_alerts_enabled           = false
   database_tunnel_alerts_enabled = false
+
+  admin_login_allowed_ip_blocks = [
```
Comment (olevett) [inline terraform/demo/main.tf:110]: I don't know if this definitely true of the resources, but various of them end up doing a sort at some point, so we might end up with unexpected changes if we don't order this - I'd be tempted to apply it and then see if a plan ends recreating any

Comment (ethanmills) [reply terraform/demo/main.tf:110]: I think this provider is slightly more intelligent: https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/wafv2/ip_set.go#L68

Comment (ethanmills) [reply terraform/demo/main.tf:110]: Given ☝️ I'm confident enough to merge this and test afterwards, fixing if necessary

**Pull request merged**